mParticleCore.brs

```diff
@@ -687,9 +694,22 @@ function mParticleStart(options as object, messagePort as object)
 
         storage.setUserAttribute = function(mpid as string, attributeKey as string, attributeValue as object) as void
             attributes = m.getUserAttributes(mpid)
+            oldValue = attributes[attributeKey]
             attributes[attributeKey] = attributeValue
             m.set(m.mpkeys.USER_ATTRIBUTES + mpid, FormatJson(attributes))
             m.flush()
+
+            if (FormatJSON(attributeValue) <> FormatJSON(oldValue)) then
+                deleted = false
+                if (attributeValue = invalid or FormatJSON(attributeValue).len() = 0) then
```

Comment:
for all of the `invalid` checks, maybe it'd be safer to user the mputils.isEmpty function?